File: lib/features/history/models/scan.dart
import 'detection.dart';

class Scan {
  final String id;
  final String imageUrl;
  final DateTime timestamp;
  final String batchId;
  final int totalBottles;
  final double totalValue;
  final List<Detection> detections;

  Scan({
    required this.id,
    required this.imageUrl,
    required this.timestamp,
    required this.batchId,
    required this.totalBottles,
    required this.totalValue,
    required this.detections,
  });

  factory Scan.fromJson(Map<String, dynamic> json) {
    return Scan(
      id: json['_id'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      timestamp: DateTime.parse(json['timestamp']),
      batchId: json['batchId'] ?? 'Unknown Batch',
      totalBottles: json['totalBottles'] ?? 0,
      totalValue: (json['totalValue'] ?? 0).toDouble(),
      detections:
          (json['detections'] as List<dynamic>?)
              ?.map((x) => Detection.fromJson(x))
              .toList() ??
          [],
    );
  }
}

File: models/Scan.js
const mongoose = require('mongoose');

const ScanSchema = new mongoose.Schema({
  imageUrl: { type: String, required: true },
  timestamp: { type: Date, default: Date.now },
  batchId: { type: String, default: "batch_default" },
  
  // Summary Stats
  totalBottles: { type: Number, required: true },
  totalValue: { type: Number, required: true },
  
  // The Detailed AI Results
  detections: [
    {
      label: String,        // "bottle", "contaminant"
      confidence: Number,
      brand: String,        // "Bisleri"
      color: String,        // "Clear", "Green"
      material: String,     // "PET", "HDPE"
      boundingBox: [Number] // [x, y, w, h]
    }
  ]
});

module.exports = mongoose.model('Scan', ScanSchema);